src/components/NcAppNavigationSearch/NcAppNavigationSearch.vue

```diff
@@ -0,0 +1,252 @@
+<!--
+  - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<docs>
+### Usage
+
+```vue
+<template>
+	<div class="styleguide-wrapper">
+		<NcContent app-name="styleguide-app-navigation" class="content-styleguidist">
+			<NcAppNavigation>
+				<template #search>
+					<NcAppNavigationSearch :value.sync="searchQuery">
+						<template #actions>
+							<NcActions aria-label="Filters">
+								<template #icon>
+									<IconFilter :size="20" />
+								</template>
+								<NcActionButton>
+									<template #icon>
+										<IconAccount :size="20" />
+									</template>
+									Filter by name
+								</NcActionButton>
+								<NcActionButton>
+									<template #icon>
+										<IconCalendarAccount :size="20" />
+									</template>
+									Filter by year
+								</NcActionButton>
+							</NcActions>
+							<NcButton aria-label="Search globally" type="tertiary">
+								<template #icon>
+									<IconSearchGlobal :size="20" />
+								</template>
+							</NcButton>
+						</template>
+					</NcAppNavigationSearch>
+				<template #list>
```

Comment:
the list is put to the search 👀

```suggestion
					</NcAppNavigationSearch>
				</template>
				<template #list>
```